src/NitroSharp/Content/ArchiveManager.cs

```diff
@@ -0,0 +1,255 @@
+    }
+
+    internal class VFSNode : IDisposable
+    {
+        public List<IArchiveFile>? MountedArchives { get; set; }
+        public Dictionary<string, VFSNode>? Children { get; set; }
+
+        private bool _disposed = false;
+
+        ~VFSNode()
+        {
+            Dispose(false);
+        }
+
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        public virtual void Dispose(bool disposing)
+        {
+            if (!_disposed)
+            {
+                if (disposing)
+                {
+                    if (Children != null)
+                    {
+                        foreach (KeyValuePair<string, VFSNode> pair in Children)
+                        {
+                            pair.Value.Dispose();
+                        }
+                    }
+
+                    if (MountedArchives != null)
+                    {
+                        foreach (IArchiveFile mountedArchive in MountedArchives)
+                        {
+                            mountedArchive.Dispose();
+                        }
+                    }
+                }
+                _disposed = true;
+            }
+        }
+    }
+
+    public class ArchiveException : Exception
+    {
+        public ArchiveException()
+            : base("Unable to open the archive")
+        {
+        }
+
+        public ArchiveException(string format, string message)
+            : base($"{format} : {message}")
+        {
+        }
+
+    }
+
+    internal class ArchiveManager : IDisposable
+    {
+        public static readonly string DefaultEncoding = "shift-jis";
+
+        private VFSNode _root;
+        private Encoding _encoding;
+
+        private bool _disposed = false;
+
+        public ArchiveManager(Encoding? encoding = null)
+        {
+            if (encoding == null)
+            {
+                try
+                {
+                    encoding = Encoding.GetEncoding(DefaultEncoding);
+                }
+                catch (ArgumentException)
+                {
+                    Encoding.RegisterProvider(CodePagesEncodingProvider.Instance);
+                    encoding = Encoding.GetEncoding(DefaultEncoding);
+                }
+            }
+            _encoding = encoding;
+            _root = new VFSNode();
+        }
+
+        ~ArchiveManager()
+        {
+            Dispose(false);
+        }
+
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        public virtual void Dispose(bool disposing)
+        {
+            if (!_disposed)
+            {
+                if (disposing)
+                {
+                    _root?.Dispose();
+                }
+                _disposed = true;
+            }
+        }
+
+        public void AddMounts(MountPoint[] mountPoints, string rootDirectory, ContentManager content)
```

Comment:
Again, I fail to see a reason to delay initialization. This is something that can be done in the constructor.

Reply:
Here the delayed initialization is required because `ArchiveManager` uses `ContentManager` to read the INI files in `system.npa`. We can bypass this by assuming that those files will always be contained in an archive but I think it's better to support hybrid situations where both archives and extracted files are used.

Reply:
Honestly I would probably just merge ``ContentManager`` and ``ArchiveManager``.